dask/layers.py

```diff
@@ -104,7 +105,26 @@ def __init__(
 #
 
 
-class SimpleShuffleLayer(Layer):
+class DataFrameLayer(Layer):
+    """DataFrame-based HighLevelGraph Layer"""
```

Comment:
One proposal in this PR is to use `DataFrameLayer` as the base class for all Dask-DataFrame-specific (columnar) Layer definitions.  The purpose is to enable more-general "column-culling" functionality in the future.  In this PR, the introduction of this class makes it possible to generalize `optimize_read_parquet_getitem` to work with CSV/ORC.  In the future,  I am hoping that we can generalize this optimization even further.